### PR TITLE
Feature/update user friend endpoint

### PR DESCRIPTION
**Pull request by lseiller**

### CODE REVIEW

File: api/src/users/users.service.ts [hunk 1]
```diff
@@ -258,4 +284,40 @@ export class UsersService {
...
+        const result = await db
+        .updateTable('user')
+        .set('lastname', updateUsersDto.lastname)
+        .where('id', '=', userId)
+        .executeTakeFirst()
+      } catch (error) {
+        throw new InternalServerErrorException();        
+      }
+    }
+  }
```
Comment (Standlc) [inline api/src/users/users.service.ts:322]: Update the table once. If necessary, create your updated object by fetching the user first, and then update. Perhaps, the `UpdateUsersDto` should have optional attributes. This way you could update like so: `.set({...updateUsersDto})`

File: api/src/migrations/migrations/2024-02-23T18:41:32.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { Kysely } from 'kysely';
+
+export async function up(db: Kysely<any>): Promise<void> {
+  await db.schema
+    .alterTable('friend')
+    .addColumn('user1_id', 'integer')
+    .addColumn('user2_id', 'integer')
```
Comment (Standlc) [inline api/src/migrations/migrations/2024-02-23T18:41:32.ts:7]: `userId`s must be a reference to an existing user:
`(col) => col.references('user.id').notNull().onDelete('cascade')`

File: api/src/users/users.service.ts [hunk 3]
```diff
@@ -157,7 +167,14 @@ export class UsersService {
...
       throw new NotFoundException();
-    return user;
+    let appUsers: AppUser[] = [];
+    user.forEach(user => {
+      appUsers.push({
+        ...user,
+        status: this.usersStatusGateway.getUserStatus(user?.id)
+      })
+    })
+    return appUsers;
```
Comment (Standlc) [inline api/src/users/users.service.ts:177]: I think a `const usersWithStatus = users.map((u) => {...u, status: userStatus})` would be cleaner.

File: api/src/users/users.service.ts (hunk 1 above)
Comment (lseiller) [reply api/src/users/users.service.ts:296]: Fix f091cbb

File: api/src/users/users.service.ts (hunk 3 above)
Comment (lseiller) [reply api/src/users/users.service.ts]: Fix 1aada99

File: api/src/friends/friends.service.ts [hunk 4]
```diff
@@ -237,41 +239,52 @@ export class FriendsService {
...
-      .select(['avatarUrl', 'id', 'username', 'rating'])
+      .select(['avatarUrl', 'id', 'username', 'rating', 'bio', 'createdAt', 'email', 'firstname', 'lastname'])
       .where('id', 'in', arrayFriendsId)
       .execute();
     } catch (error) {
       console.log(error);
       throw new InternalServerErrorException();
     }
-    return friendList;
+    return friendList.map(u => ({...u, status: this.usersStatusGateway.getUserStatus(u?.id)}))
```
Comment (Standlc) [inline api/src/friends/friends.service.ts:287]: Pretty sure this works:
``` 
const friends = await db
      .selectFrom('user')
      .where('user.id', '!=', userId)
      .innerJoin('friend', (join) =>
        join.on((eb) =>
          eb.or([
            eb('friend.user1_id', '=', eb.ref('user.id')),
            eb('friend.user2_id', '=', eb.ref('user.id')),
          ]),
        ),
      )
      .select([
        'user.avatarUrl',
        'user.id',
        'user.username',
        'user.rating',
        'user.bio',
        'user.createdAt',
        'user.email',
        'user.firstname',
        'user.lastname',
      ])
      .execute();
    return friends.map((u) => ({
      ...u,
      status: this.usersStatusGateway.getUserStatus(u?.id),
    }));
```

**Review by AlexSokolov9 — APPROVED** [review]

**Pull request merged**